File: src/main/javascript/algo/search/binary_search.js
function binarySearch(array, key) {
  let left = 0;
  let right = array.length - 1;

  while (left <= right) {
    const mid = left + Math.floor((right - left) / 2);

    if (array[mid] === key) {
      return mid;
    }

    if (array[mid] < key) {
      left = mid + 1;
    } else {
      right = mid - 1;
    }
  }

  return -1;
}

function verify(index) {
  if (index === -1) {
    console.log('key not found in array');
  } else {
    console.log(`Key found at index: ${index}`);
  }
}

let result;
const numberArr = [1, 2, 3, 4, 5, 6];

result = binarySearch(numberArr, 8);
verify(result);

result = binarySearch(numberArr, 5);
verify(result);

File: src/main/java/algo/search/BinarySearch.java
package main.java.algo.search;

// Iterative
public class BinarySearch {

    public static int binarySearch(int[] arr, int target) {
        int first = 0;
        int last = arr.length - 1;

        while(first <=last){
            int mid = (first + last) / 2;

            if(arr[mid] == target) {
                return mid;
            } else if(arr[mid] < target) {
                first = mid + 1;
            } else {
                last = mid - 1;
            }
        }

        return -1;
    }

    public static void verify(int index) {
        if (index == -1) {
            System.out.println("Key was not found in the array");
        } else {
            System.out.println("Key found at index: " + index);
        }
    }

    public static void main(String[] args) {

        int[] arr = {1,2,3,4,5,6,7,8,9,10};

        int result = binarySearch(arr, 12);
        verify(result);

        result = binarySearch(arr, 5);
        verify(result);

    }

}
